test/browser/anchor-manager.test.jsx

```diff
@@ -0,0 +1,261 @@
+    anchors.splice(anchors.indexOf(anchor, 1));
+  };
+
+  beforeEach(function() {
+    clock = sinon.useFakeTimers();
+    manager = StubAnchorManager();
+    anchors = [];
+  });
+
+  afterEach(function() {
+    anchors.slice().forEach(unmountAnchor);
+    anchors.length.should.equal(0);
+    clock.restore();
+  });
+
+  it('should not do anything if anchorId is not defined', function() {
+    renderAnchor();
+    Object.keys(manager.anchorMap).should.eql([]);
+    manager.initialized.should.be.false;
+  });
+
+  it('should register w/ anchor manager if anchorId is defined', function() {
+    renderAnchor({anchorId: 'foo'});
+    Object.keys(manager.anchorMap).should.eql(['foo']);
+  });
+
+  it('should unregister w/ anchor manager on unmount', function() {
+    var a = renderAnchor({anchorId: 'foo'});
+    unmountAnchor(a);
+    Object.keys(manager.anchorMap).should.eql([]);
+  });
+
+  it('should use singleton anchor manger by default', function() {
+    var m1 = SampleAnchorClass.getDefaultProps().anchorManager;
+    var m2 = SampleAnchorClass.getDefaultProps().anchorManager;
+
+    m1.should.be.an.Object;
+    m1.should.equal(m2);
+  });
+
+  it('logs warning if props.anchorId changes', function() {
+    var a = renderAnchor({anchorId: 'foo'});
+    sinon.stub(console, 'warn');
+    a.componentDidUpdate({anchorId: 'bar'});
+    console.warn.callCount.should.equal(1);
+    console.warn.getCall(0).args.should.eql([
+      '"anchorId" prop is expected to be constant, but changed.'
+    ]);
+    console.warn.restore();
+  });
+
+  it('cancels attract timeout on unmount', function() {
+    var a = renderAnchor({anchorId: 'foo'});
+    sinon.spy(a, 'cancelAttractAttentionToAnchor');
+    a.attractAttentionToAnchor();
+    a.cancelAttractAttentionToAnchor.callCount.should.equal(0);
+    unmountAnchor(a);
+    a.cancelAttractAttentionToAnchor.callCount.should.equal(1);
+    clock.tick(999999999);
+    a.cancelAttractAttentionToAnchor.callCount.should.equal(1);
+    a.cancelAttractAttentionToAnchor.restore();
+  });
+
+  it('does not attract attention to anchor when mounted', function() {
+    var a = renderAnchor({anchorId: 'foo'});
+    a.state.attractAttentionToAnchor.should.be.false;
+  });
+
+  it('attracts attention to anchor when navigated to', function() {
+    var a = renderAnchor({anchorId: 'foo'});
+    manager.simulateHashChange('foo');
+    a.state.attractAttentionToAnchor.should.be.true;
+  });
+
+  it('stops attracting attention to anchor after some time', function() {
+    var a = renderAnchor({anchorId: 'foo'});
+    manager.simulateHashChange('foo');
+    clock.tick(AnchorManagerMixin.DEFAULT_ATTRACT_DURATION + 1);
+    a.state.attractAttentionToAnchor.should.be.false;
+  });
+
+  it('waits ATTRACT_ATTENTION_TO_ANCHOR_DURATION if defined', function() {
+    var a = renderAnchor({anchorId: 'foo'});
+    a.ATTRACT_ATTENTION_TO_ANCHOR_DURATION = 10;
+    a.attractAttentionToAnchor();
```

Comment:
What do you think about adding a `a.state.attractAttentionToAnchor.should.be.true;` check here? @toolness